lib/auth/bot.go

```diff
@@ -98,11 +105,19 @@ func createBotUser(ctx context.Context, s *Server, botName string, resourceName
 	user.SetMetadata(metadata)
 
 	// Traits need to be set to silence "failed to find roles or traits" warning
-	user.SetTraits(map[string][]string{
-		constants.TraitLogins:     {},
-		constants.TraitKubeUsers:  {},
-		constants.TraitKubeGroups: {},
-	})
+	if traits == nil {
+		traits = map[string][]string{}
+	}
+	if _, ok := traits[constants.TraitLogins]; !ok {
+		traits[constants.TraitLogins] = []string{}
+	}
+	if _, ok := traits[constants.TraitKubeUsers]; !ok {
+		traits[constants.TraitKubeUsers] = []string{}
+	}
+	if _, ok := traits[constants.TraitKubeGroups]; !ok {
+		traits[constants.TraitKubeGroups] = []string{}
+	}
```

Comment:
Should we add other traits here as well then? E.g. database users, database names, desktop logins. If we decided to add them to bot traits in future.

Reply:
I'm not awfully sure. It seems odd to me that I actually need to initialise these empty slices, but I wanted to ensure I didn't change the existing behaviour which is apparently to prevent a `"failed to find roles or traits"` warning. Looking into this more thoroughly, it looks https://github.com/gravitational/teleport/pull/10673 actually means this is no longer necessary.

Reply:
https://github.com/gravitational/teleport/pull/15117/commits/3fd36bb963bc24174f6e3359c7f669bde57239e2